pilot/pkg/model/context.go

```diff
@@ -199,6 +202,17 @@ func (node *Proxy) SetServiceInstances(env *Environment) error {
 	return nil
 }
 
+func (node *Proxy) SetWorkloadLabels(env *Environment) error {
+	labels, err := env.GetProxyWorkloadLabels(node)
+	if err != nil {
+		log.Warnf("failed to get service proxy workload labels: %v, defaulting to proxy metadata", err)
+		labels = LabelsCollection{node.Metadata}
```

Comment:
@robertpanzer this is not the right way to collect labels.. all the node labels are under the metadata with ISTIO_META_LABELS_... prefix or something like that.  If you simply use all metadata, it will accumulate a lot of junk.

Reply:
I would have to double check, but I think node.Metadata contains k8s labels, k8s annotations, and ISTIO_META_LABELS_... but with the `ISTIO_META_LABELS_` prefix already stripped?

Reply:
It also contains non label stuff like HTTP10, interception_mode, etc.. doesn't it?

Reply:
Yeah since it pulls in the annotations. it is pretty confusing what gets thrown in there and the code isn't easy to follow, we may need to clean it up/test it/document it later

Reply:
I just debugged into the code, and the labels – at least for me – are flat key values pairs in node.Metadata.
Also it doesn't look like there's anything stripped:
![Screen Shot 2019-04-10 at 6 14 18 PM](https://user-images.githubusercontent.com/1163662/55895496-913a3100-5bbc-11e9-917c-eedd0eb7a0d2.png)
To me it's just the plain Metadata map that is in the Node.Metadata of the Envoy configuration. (So the name really makes sense)

Reply:
![image](https://user-images.githubusercontent.com/8202871/55897151-f6c2f900-5b8d-11e9-8ed2-6750eed9430c.png)

where are these going?

Reply:
never mind. they are all flattened into one big kv struct.. this is not good. but its not your problem either. So, lets go with whatever you currently have